metadeploy/api/tests/jobs.py

```diff
@@ -346,6 +346,86 @@ def test_create_scratch_org(self, settings, plan_factory, scratch_org_factory):
+        plan = plan_factory()
+        with ExitStack() as stack:
+            stack.enter_context(patch("metadeploy.api.jobs.local_github_checkout"))
+            jwt_session = stack.enter_context(
+                patch("metadeploy.api.salesforce.jwt_session")
+            )
+            jwt_session.return_value = {
+                "instance_url": "https://sample.salesforce.org/",
+                "access_token": "abc123",
+                "refresh_token": "abc123",
+            }
+            open = stack.enter_context(patch("metadeploy.api.salesforce.open"))
+            fake_json = json.dumps({"edition": ""})
+            open.return_value = MagicMock(
+                **{
+                    "__enter__.return_value": MagicMock(
+                        **{"read.return_value": fake_json}
+                    )
+                }
+            )
+            SimpleSalesforce = stack.enter_context(
+                patch("metadeploy.api.salesforce.SimpleSalesforce")
+            )
+            SimpleSalesforce.return_value = MagicMock(
+                **{
+                    "ScratchOrgInfo.get.return_value": {
+                        "LoginUrl": "https://sample.salesforce.org/",
+                        "ScratchOrg": "abc123",
+                        "SignupUsername": "test",
+                        "AuthCode": "abc123",
+                    }
+                }
+            )
+            SalesforceOAuth2 = stack.enter_context(
+                patch("metadeploy.api.salesforce.SalesforceOAuth2")
+            )
+            SalesforceOAuth2.return_value = MagicMock(
+                **{
+                    "get_token.return_value": MagicMock(
+                        **{
+                            "json.return_value": {
+                                "access_token": "abc123",
+                                "refresh_token": "abc123",
+                            }
+                        }
+                    )
+                }
+            )
+            BaseCumulusCI = stack.enter_context(
+                patch("metadeploy.api.salesforce.BaseCumulusCI")
+            )
+            org_config = MockDict()
+            org_config.config_file = "/"
+            BaseCumulusCI.return_value = MagicMock(
+                **{
+                    "project_config.repo_root": "/",
+                    "keychain.get_org.return_value": org_config,
+                }
+            )
+            stack.enter_context(patch("metadeploy.api.salesforce.DeployOrgSettings"))
+            stack.enter_context(
+                patch("cumulusci.core.flowrunner.PreflightFlowCoordinator.run")
+            )
+            # Cheat the auto-triggering of the job by adding a fake
+            # enqueued_at:
+            scratch_org = scratch_org_factory(
+                plan=plan,
+                enqueued_at=datetime(2020, 9, 4, 12),
+            )
+            stack.enter_context(patch("metadeploy.api.jobs.job_started"))
```

Comment:
@wlonk I'm not entirely sure why this patch is needed. Maybe something related to the atomic transaction? Without it I was getting an error because it seems the plan slug or plan template was not created...

```
_____________________________________________________________________________________________ TestCreateScratchOrg.test_create_scratch_org__no_preflight ______________________________________________________________________________________________
/usr/local/lib/python3.8/site-packages/django/db/models/fields/related_descriptors.py:164: in __get__
    rel_obj = self.field.get_cached_value(instance)
/usr/local/lib/python3.8/site-packages/django/db/models/fields/mixins.py:13: in get_cached_value
    return instance._state.fields_cache[cache_name]
E   KeyError: 'plan_template'

During handling of the above exception, another exception occurred:
metadeploy/api/tests/jobs.py:423: in test_create_scratch_org__no_preflight
    create_scratch_org(
metadeploy/api/jobs.py:324: in create_scratch_org
    async_to_sync(job_started)(org, job)
/usr/local/lib/python3.8/site-packages/asgiref/sync.py:139: in __call__
    return call_result.result()
/usr/local/lib/python3.8/concurrent/futures/_base.py:432: in result
    return self.__get_result()
/usr/local/lib/python3.8/concurrent/futures/_base.py:388: in __get_result
    raise self._exception
/usr/local/lib/python3.8/site-packages/asgiref/sync.py:204: in main_wrap
    result = await self.awaitable(*args, **kwargs)
metadeploy/api/push.py:223: in job_started
    "plan_slug": plan.slug,
/src/sfdo-template-helpers/sfdo_template_helpers/slugs.py:50: in slug
    slug = self.slug_queryset.filter(is_active=True).first()
metadeploy/api/models.py:482: in slug_queryset
    return self.plan_template.planslug_set
/usr/local/lib/python3.8/site-packages/django/db/models/fields/related_descriptors.py:178: in __get__
    rel_obj = self.get_object(instance)
/usr/local/lib/python3.8/site-packages/django/db/models/fields/related_descriptors.py:145: in get_object
    return qs.get(self.field.get_reverse_related_filter(instance))
/usr/local/lib/python3.8/site-packages/django/db/models/query.py:406: in get
    raise self.model.DoesNotExist(
E   metadeploy.api.models.PlanTemplate.DoesNotExist: PlanTemplate matching query does not exist.
```